### DISCUSSION TITLE
Bonsai + DLClive coordinates to measure distance

### DISCUSSION THREAD

**Discussion by MilanNedoma** [bonsai-rx/bonsai#1587] [Q&A] [answered]
Hi!

I am trying to get the use DLC live in Bonsai and I have stumbled upon a problem. I wasn´t successful with finding a way to get the XY coordinates out of the PredictPose node to compare it with other XY coordinates and started to work on a python script that would allow me to do so. I have tried this so far, however, I keep getting an error stating: "Unexpected token" <newline>. I have looked this up, however, I am quite new to this and as far as I can tell, there´s not a very detailed documentation regarding DLC live and Bonsai use together. 
Please, see the code attached below and let me know if you would know how to solve this:

```
import System;
import System.Reactive.Linq;
import Bonsai;
import Bonsai.Dlc;

[Combinator]
[WorkflowElementCategory(ElementCategory.Source)]
class ExtractDlcCoordinates
{
 public IObservable<PointFrame> Process(IObservable<PointFrame> source)
    {
        return source.Select(frame =>
        {
            var xCoordinate = frame.MarbleCentre.X;
            var yCoordinate = frame.MarbleCentre.Y;
            Console.WriteLine($"X Coordinate: {xCoordinate}, Y Coordinate: {yCoordinate}");
            return frame;
        });
    }
}
  
```

Thank you very much in advance.

**Comment by bruno-f-cruz** [comment]
Can you share your workflow?

**Comment by MilanNedoma** [comment]
Hey,

sure, please see the file attached below.
I am also attaching the screenshot of the workflow.
[Workflow.txt](https://github.com/bonsai-rx/bonsai/files/12878230/Workflow.txt)

![image](https://github.com/bonsai-rx/bonsai/assets/140824061/3b9f7e47-56f6-4b5f-a790-a9c072c49913)

**Reply by bruno-f-cruz** [reply, marked as answer]
As described in [the paper](https://elifesciences.org/articles/61909), you should use the operator "GetBodyPart" to get the information relative to a particular bodypart of the pose. Once you do that you will have access to the raw coordinate from the prediction.

**Reply by MilanNedoma** [reply]
Thank you very much for this, it does solve my primary issue.
However, I am trying to set up a looming test with this pipeline and I am still struggling to find a way to activate a looming stimuli when a mouse is going to cross a certain boundary of the X axis.  As you can see, I have tried to convert the int to a float to match both datatypes and then used the "GreaterOrEqual" node, which I hoped to use to simply activate the stimulus afterwards.
I have also tried reading the documentation and I was not able to find whether it is a problem to use float values for this node, but I was not successful in finding anything that would suggest this. Whenever I try to run it at this point, I get a message from the Node saying "Unsupported number of arguments. This node supports at most 1 input connection(s)". Whilst this being totally easy fix, I do not understand the nature of the message. I have seen other workflows putting two inputs to this node and everything worked fine, not mentioning that it makes sense to me to put two inputs on order to compare them.
If I am mistaken in this, please, do let me know, I am not very experienced in this regard and I am very much willing to learn!

I am sending you my updated workflow folder and a screenshot: 
[Looming.txt](https://github.com/bonsai-rx/bonsai/files/12916752/Looming.txt)

![image](https://github.com/bonsai-rx/bonsai/assets/140824061/830b9ad0-2055-4990-8517-476c3c23ebbf)

Thank you very much in advance.

**Comment by bruno-f-cruz** [comment]
That node doesn’t have an overload that takes two separate sequences as
inputs, I think you are might be referring to a situation where you create
a tuple, for instance, using combinelatest on those two sequences before
the comparison operator. Can you send me a link to the post you are
referring to?

**Reply by MilanNedoma** [reply]
This is the post I am talking about, just from the slight glimpse of the screenshot, I saw that the nodes are used in the way I have mentioned previously:
https://groups.google.com/g/bonsai-users/c/fINACqCkdB8/m/MQECV_eRAwAJ

Coming back to what you have mentioned, after simply putting in the "CombineLatest" node, not much has improved. 
![image](https://github.com/bonsai-rx/bonsai/assets/140824061/ee3a6eb7-399d-4d74-853b-7c0c58f96333)
Now, when trying to run the pipeline, it shows a message saying: "Binary operator cannot be applied to input arguments when a default operand is already specified". Does this have to do anything with the fact that the GreaterThanOrEqual node has the option to set its own values?

**Reply by bruno-f-cruz** [reply]
Can you remove the comparison operator and add it again?

**Reply by MilanNedoma** [reply]
Worked flawlessy, thank you very much!
